### PR TITLE
Adding test coverage

### PR DESCRIPTION
**Pull request by Jordybastien**
### What this PR is doing?

1. Adding test coverage report
2. Adding Code climate & Pushing the coverage report to code climate
3. Adding badges(maintainability & test coverage) to repository readMe

We have some work😅
Screenshot 2024-05-16 at 08 34 58

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -37,6 +37,13 @@ jobs:
         run: bin/rails db:schema:load
       - name: Run tests
         run: bin/rake test
+      - name: Push test coverage to Code Climate
+        uses: paambaati/codeclimate-action@v6.0.0
+        env:
+          CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
+        with:
+          coverageCommand: bin/rake test
```
Comment (hoglund003) [inline .github/workflows/main.yml:45]: Is this doing the same as the first command, or is it JUST for the coverage page?: `run: bin/rake test`

**Review by hoglund003 — APPROVED** [review]
Very cool! :star_struck:

**Review by str1fe — APPROVED** [review]
Nice!

File: .github/workflows/main.yml (hunk 1 above)
Comment (Jordybastien) [reply .github/workflows/main.yml:43]: Great catch that was actually redundant, it was running the tests all over again then pushing the report.

I've combined both

**Pull request merged**